### PR TITLE
Move Sortable List into its own package

### PR DESCRIPTION
**Pull request by hprateek43**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
#71863 is an active open request for scheduler cleanup. In continuation with that, I have removed the sortableList package and replaced it by native golang slice.
**Which issue(s) this PR fixes**:

Part of #71863

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by hprateek43** [comment]
@draveness Can you give ok-to-test

File: pkg/scheduler/util/utils.go [hunk 1]
```diff
@@ -92,30 +91,6 @@ func GetEarliestPodStartTime(victims *extenderv1.Victims) *metav1.Time {
 	return earliestPodStartTime
 }
 
-// SortableList is a list that implements sort.Interface.
```
Comment (ahg-g) [inline pkg/scheduler/util/utils.go]: we don't need to create our own "SortableList", we can use [sort.Slice](https://golang.org/pkg/sort/#Slice).

File: pkg/scheduler/core/generic_scheduler.go [hunk 2]
```diff
@@ -1089,7 +1090,7 @@ func (g *genericScheduler) selectVictimsOnNode(
 	if nodeInfo == nil {
 		return nil, 0, false
 	}
-	potentialVictims := util.SortableList{CompFunc: util.MoreImportantPod}
+	potentialVictims := sortablelist.SortableList{CompFunc: util.MoreImportantPod}
```
Comment (ahg-g) [inline pkg/scheduler/core/generic_scheduler.go]: create a normal slice and use just use ```slice.Sort(potentialVictims, util.MoreImportantPod)``` to sort.

**Review by draveness — COMMENTED** [review]
/ok-to-test

**Review by mattjmcnaughton — COMMENTED** [review]
Thanks for undertaking this clean up :)

I'm +1 with @ahg-g - if we have the option to delete `SortableList` entirely, and instead use go builtins, that appears superior. Are there any advantages to `SortableList` that I'm missing?

File: pkg/scheduler/core/generic_scheduler.go (hunk 2 above)
Comment (hprateek43) [reply pkg/scheduler/core/generic_scheduler.go]: util. MoreImportantPod expects an `interface{}` while sort.Slice expects an `int` in its sorting function. Should I change the logic of more important pod to solve this?

File: pkg/scheduler/util/utils.go (hunk 1 above)
Comment (hprateek43) [reply pkg/scheduler/util/utils.go]: I have made the change. Can you review it once?

File: pkg/scheduler/core/generic_scheduler.go (hunk 2 above)
Comment (hprateek43) [reply pkg/scheduler/core/generic_scheduler.go]: I have made the change. Can you review it once?

File: pkg/kubelet/util/sliceutils/sliceutils.go [hunk 3]
```diff
@@ -32,7 +32,7 @@ func StringInSlice(s string, list []string) bool {
 	return false
 }
 
-// PodsByCreationTime makes an array of pods sortable by their creation
```
Comment (ahg-g) [inline pkg/kubelet/util/sliceutils/sliceutils.go]: please revert the changes in this file, we don't want to wait for a kubelet review.

File: pkg/printers/internalversion/printers.go [hunk 4]
```diff
@@ -2114,7 +2114,7 @@ func printBool(value bool) string {
 	return "False"
 }
 
-// SortableResourceNames - An array of sortable resource names
```
Comment (ahg-g) [inline pkg/printers/internalversion/printers.go]: ditto, revert this.

File: pkg/scheduler/util/utils_test.go [hunk 5]
```diff
@@ -50,14 +52,15 @@ func TestSortableList(t *testing.T) {
 				Priority: &p,
 			},
 		}
-		podList.Items = append(podList.Items, pod)
+		podList = append(podList, pod)
 	}
-	podList.Sort()
-	if len(podList.Items) != 10 {
-		t.Errorf("expected length of list was 10, got: %v", len(podList.Items))
+	sort.Slice(podList, func(i, j int) bool { return higherPriority(podList[i], podList[j]) })
```
Comment (ahg-g) [inline pkg/scheduler/util/utils_test.go]: instead of defining the function, just use the implementation here and delete the definition in line 36:

```
sort.Slice(podList, func(i, j int) bool { pod.GetPodPriority(pod1) > pod.GetPodPriority(pod2) })
```

also change ```podList``` type to ```v1.Pod``` instead of ```interface{}``` in line 40

File: pkg/scheduler/core/generic_scheduler.go [hunk 6]
```diff
@@ -1105,7 +1105,7 @@ func (g *genericScheduler) selectVictimsOnNode(
 	queue internalqueue.SchedulingQueue,
 	pdbs []*policy.PodDisruptionBudget,
 ) ([]*v1.Pod, int, bool) {
-	potentialVictims := util.SortableList{CompFunc: util.MoreImportantPod}
+	var potentialVictims []interface{}
```
Comment (ahg-g) [inline pkg/scheduler/core/generic_scheduler.go]: use *v1.Pod as the type.

File: pkg/scheduler/core/generic_scheduler.go [hunk 7]
```diff
@@ -1161,11 +1161,11 @@ func (g *genericScheduler) selectVictimsOnNode(
 	}
 	var victims []*v1.Pod
 	numViolatingVictim := 0
-	potentialVictims.Sort()
+	sort.Slice(potentialVictims, func(i, j int) bool { return util.MoreImportantPod(potentialVictims[i], potentialVictims[j]) })
```
Comment (ahg-g) [inline pkg/scheduler/core/generic_scheduler.go]: change ```MoreImportantPod``` to accept ```*v1.Pod``` instead of ```interface{}```

**Review by ahg-g — COMMENTED** [review]
Thanks a lot for doing this! few more comments to cleanup more redundant operations.

File: pkg/scheduler/util/utils_test.go [hunk 8]
```diff
@@ -32,10 +33,8 @@ import (
 // TestSortableList tests SortableList by storing pods in the list and sorting
 // them by their priority.
 func TestSortableList(t *testing.T) {
-	higherPriority := func(pod1, pod2 interface{}) bool {
-		return pod.GetPodPriority(pod1.(*v1.Pod)) > pod.GetPodPriority(pod2.(*v1.Pod))
-	}
-	podList := SortableList{CompFunc: higherPriority}
+
```
Comment (ahg-g) [inline pkg/scheduler/util/utils_test.go]: This test isn't necessary now since we there is no "sortablelist" type anymore.

**Comment by ahg-g** [comment]
Since we removed sortablelist, please remove the corresponding unit test as well.

**Comment by hprateek43** [comment]
> Since we removed sortablelist, please remove the corresponding unit test as well.

Made the change. Can your review?

**Comment by ahg-g** [comment]
Update bazel

**Comment by hprateek43** [comment]
@ahg-g Can you add a `lgtm` ?
